tests/sentry/api/endpoints/test_monitor_details.py

```diff
@@ -1,135 +1,197 @@
 from datetime import timedelta
 
+from django.urls import reverse
 from django.utils import timezone
 
 from sentry.models import Monitor, MonitorStatus, MonitorType, ScheduledDeletion, ScheduleType
 from sentry.testutils import APITestCase
+from sentry.testutils.silo import region_silo_test
 
 
+@region_silo_test(stable=True)
 class MonitorDetailsTest(APITestCase):
-    def test_simple(self):
-        user = self.create_user()
-        org = self.create_organization(owner=user)
-        team = self.create_team(organization=org, members=[user])
-        project = self.create_project(teams=[team])
-
-        monitor = Monitor.objects.create(
-            organization_id=org.id,
-            project_id=project.id,
+    def setUp(self):
+        self.user = self.create_user()
+        self.org = self.create_organization(owner=self.user)
+        self.team = self.create_team(organization=self.org, members=[self.user])
+        self.project = self.create_project(teams=[self.team])
+
+        self.monitor = Monitor.objects.create(
+            organization_id=self.org.id,
+            project_id=self.project.id,
             next_checkin=timezone.now() - timedelta(minutes=1),
             type=MonitorType.CRON_JOB,
             config={"schedule": "* * * * *"},
         )
 
-        self.login_as(user=user)
+        self.paths = (
+            f"/api/0/monitors/{self.monitor.guid}/",
+            f"/api/0/monitors/{self.org.slug}/{self.monitor.guid}/",
+        )
+
+    def test_simple(self):
+        self.login_as(user=self.user)
+
         with self.feature({"organizations:monitors": True}):
-            resp = self.client.get(f"/api/0/monitors/{monitor.guid}/")
+            for path in self.paths:
+                resp = self.client.get(path)
+
+                assert resp.status_code == 200, resp.content
+                assert resp.data["id"] == str(self.monitor.guid)
 
-        assert resp.status_code == 200, resp.content
-        assert resp.data["id"] == str(monitor.guid)
+    def test_mismatched_org_slugs(self):
+        path = f"/api/0/monitors/asdf/{self.monitor.guid}/"
+        self.login_as(user=self.user)
 
+        with self.feature({"organizations:monitors": True}):
+            resp = self.client.get(path)
 
+            assert resp.status_code == 400
+
+
+@region_silo_test(stable=True)
 class UpdateMonitorTest(APITestCase):
     def setUp(self):
         self.user = self.create_user()
         self.org = self.create_organization(owner=self.user)
         self.team = self.create_team(organization=self.org, members=[self.user])
         self.project = self.create_project(teams=[self.team])
+        self.num_paths = 2
```

Comment:
Nit: if you don't go with my lambda-based suggestion, this could be `len(self._get_urls())`.